docs/technical-overview.md

```diff
@@ -0,0 +1,267 @@
+*[➡️ Documentation index](./index.md)*
+
+# Technical Overview
+
+**Table of contents**
+
+* [Fixtures](#fixtures)
+* [Manufacturers](#manufacturers)
+* [Fixture model](#fixture-model)
+* [Fixture features](#fixture-features)
+* [Plugins](#plugins)
+  * [Exporting](#exporting)
+  * [Importing](#importing)
+  * [Export Tests](#export-tests)
+* [Testing](#testing)
+* [UI / Website](#ui--website)
+
+## Fixtures
+*[⬆️ Back to top](#technical-overview)*
+
+A fixture (sometimes also *personality*) is a lighting device that can be controlled with DMX. OFL gathers fixture definitions, that are specifications of a fixture's general information (like physical data) and the details of its [DMX](https://www.learnstagelighting.com/what-is-dmx-512/) controlment.
+
+Each fixture belongs to a [manufacturer](#manufacturers) (many-to-one relationship). A manufacturer is the vendor or brand of the fixture.
+
+The fixtures are saved as [JSON](http://www.json.org/) files at `fixtures/<manufacturer-key>/<fixture-key>.json`. The fixture key is only defined by the filename. See [details about the fixture JSON format](TODO).
+
+The JSON fixture data is parsed and processed using our [model](#fixture-model).
+
+## Manufacturers
+*[⬆️ Back to top](#technical-overview)*
+
+A manufacturer is a [fixture](#fixtures) vendor or brand. Each fixture belongs to a manufacturer (0-n fixtures to 1 manufacturer relationship).
+
+All used manufacturers must be defined in `fixtures/manufacturers.json` in order that their unique keys can be registered. All fixtures of a manufacturer are saved in the `fixtures/<manufacturer-key>/` directory.
+
+We also store (optional) additional manufacturer data like comment, website or [RDM](TODO) data.
+
+## Fixture model
+*[⬆️ Back to top](#technical-overview)*
+
+Instead of parsing a [fixture](#fixtures)'s JSON data directly, it is recommended to use the model. We developed it to easier handle complicated fixture features like fine/switching channels.
+
+The model uses [ES2015 classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) to represent the fixtures. E. g., `Fixture.fromRepository('cameo', 'nanospot-120')` returns an instance of the `Fixture` class, instantiated with the specified fixture's data. These objects have several convenient properties that allow an easy usage of the fixture data in [plugins](#plugins), [UI](#ui--website) and more.
```

Comment:
* "... returns a `Fixture` object, instanciated ..." to avoid repetition
* "... allow ~an~ easy usage ..."